### PR TITLE
Chore/self hosted assistant disabled

### PR DESCRIPTION
**Pull request by saltcod**
in self-hosted, if you don't have a OPENAI_API_KEY set, the assistant will look like it's working, but will just silently fail.

added an api route to check for the existence of the key, and an alert to notify if not present

screenshot-2024-10-30-at-16 40 00

### CODE REVIEW

File: apps/studio/components/interfaces/SQLEditor/AiAssistantPanel/index.tsx
Comment (joshenlim) [inline apps/studio/components/interfaces/SQLEditor/AiAssistantPanel/index.tsx:137]: this could probably be a react query - will be cleaner i think

File: apps/studio/pages/api/ai/sql/check-api-key.ts
Comment (joshenlim) [inline apps/studio/pages/api/ai/sql/check-api-key.ts:9]: this doesn't seem to work (running into 500) and is also not consistent with the way we write our next js endpoints

File: apps/studio/components/interfaces/SQLEditor/AiAssistantPanel/index.tsx
Comment (joshenlim) [reply apps/studio/components/interfaces/SQLEditor/AiAssistantPanel/index.tsx:137]: also thinking - probably doesnt need to be self hosted specific? this check should also work for hosted too, probably a good safety net for us actually haha

**Comment by joshenlim** [comment]
Pushed up the fixes for the issues identified
Also added the same check to the global assistant UI

File: apps/studio/components/interfaces/SQLEditor/AiAssistantPanel/index.tsx
Comment (joshenlim) [reply apps/studio/components/interfaces/SQLEditor/AiAssistantPanel/index.tsx]: ^ actually sorry think if we do this API call on the hosted dash it'll probably incur unnecessary functions cost

**Review by joshenlim — APPROVED** [review]
LGTM! Tested locally for if API key is not set, and on staging preview for hosted
@saltcod will let you merge whenever ready - probably good to double check the changes i made here before merging too 🙏

**Comment by saltcod** [comment]
amazing, thanks @joshenlim! Checked staging and local, merging.

**Pull request merged**
